### PR TITLE
added executor openapi spec

### PR DESCRIPTION
**Pull request by exu**
This PR...

## Changes

-

## Fixes

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Comment by exu** [comment]
@olensmar it would be great to have somehow openapi spec generated for this one - but no website here any idea ? maybe some link in docs?

File: api/executor.yaml [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+    Execution:
+      description: execution result returned from executor
+      type: object
+      properties:
+        id:
+          type: string
+          description: execution id
+          format: bson objectId
+        script-content:
```
Comment (olensmar) [inline api/executor.yaml:106]: let's do camelCase for properties

File: api/executor.yaml [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+
+#
+# Requests
+#
+
+    ExecutionRequest: 
+      description: scripts execution request body
+      type: object
+      properties:
```
Comment (olensmar) [inline api/executor.yaml:215]: add required properties (goes for all objects)?

File: api/executor.yaml [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+externalDocs:
+  description: 'Find out more about kubtest'
+  url: http://kubtest.io
+
+tags:
+- name: executor
+  description: 'scripts execution API' 
+
+paths:
+  /executions/:
```
Comment (olensmar) [inline api/executor.yaml:22]: remove trailing slash

File: api/executor.yaml [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+        400:
+          description: 'problem with request body'
+          content:
+            application/problem+json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Problem'
+        500:
+          description: 'poblem with saving script details into queue'
```
Comment (olensmar) [inline api/executor.yaml:54]: poblem -> Problem

File: api/executor.yaml [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+        - in: path
+          name: id
+          schema:
+            type: string
+          required: true
+          description: 'ID of the execution'
+      tags:
+      - executor
+      summary: 'Get execution by ID'
+      description: 'Gets execution with given ID, returns deailed information about test execution'
```
Comment (olensmar) [inline api/executor.yaml:74]: deailed -> detailed

**Review by olensmar — CHANGES_REQUESTED** [review]
see inline comments

**Pull request closed without merging**